### PR TITLE
Filebrowser helm chart

### LINKED ISSUE #12: Helm charts for meetings and office
to be included in the upcoming version of the marketplace over VDCs

it's required to have
- [x] meetings chart
- [x] office chart

related`;

- [ ] icon for filebrowser https://github.com/threefoldtech/vdc-solutions-charts/issues/32

### PR DESCRIPTION
**Pull request by tobiaschielens**

### CODE REVIEW

**Review by m-motawea — CHANGES_REQUESTED** [review]
please add `readme` to explain the required config for the chart

you can use https://github.com/threefoldtech/vdc-solutions-charts/blob/master/charts/kube-prometheus-stack/README.md as an example

File: charts/filebrowser/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
```
Comment (ashraffouda) [inline charts/filebrowser/templates/deployment.yaml:70]: plz refer to this [README.md](https://github.com/threefoldtech/vdc-solutions-charts/blob/master/README.md#note) notes section and add limits to the charts

File: charts/filebrowser/templates/hpa.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        name: cpu
+        targetAverageUtilization: {{ .Values.autoscaling.targetCPUUtilizationPercentage }}
+    {{- end }}
+    {{- if .Values.autoscaling.targetMemoryUtilizationPercentage }}
+    - type: Resource
+      resource:
+        name: memory
+        targetAverageUtilization: {{ .Values.autoscaling.targetMemoryUtilizationPercentage }}
+    {{- end }}
+{{- end }}
```
Comment (ashraffouda) [inline charts/filebrowser/templates/hpa.yaml:28]: plz refer to this [README.md](https://github.com/threefoldtech/vdc-solutions-charts/blob/master/README.md#note) notes section and add limits to the charts

**Review by ashraffouda — CHANGES_REQUESTED** [review]

**Comment by ashraffouda** [comment]
all deployments are missing the resources limits part, commented on two of them but other deployments needs that too
plz refer to this [README.md](https://github.com/threefoldtech/vdc-solutions-charts/blob/master/README.md#note) notes section and add limits to the charts

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
meeting chart in another PR, Can we remove it from this one?

can you please follow these [steps](https://github.com/threefoldtech/vdc-solutions-charts#addingmodifying-a-chart-in-the-repo) for adding the chart?

**Comment by tobiaschielens** [comment]
Added readme + removed meetings + added resource limits

File: charts/filebrowser/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+This is a basic helm chart that allows to run Filebrowser full node on a Kubernetes cluster.
+
+Filebrowser can be deployed using the following steps:
+helm install filebrowser ./ \
+--set DOCUMENTSERVER_URL="URL to the documentserver filebrowser can use"
+--set THREEBOT_APP_ID="Hostname where the application will be exposed" 
+
+Important note:
+Make sure to expose the deployed solution on default webserver ports  (80 or 443)
```
Comment (ashraffouda) [inline charts/filebrowser/README.md]: can u plz write detailed steps in readme about what u mean with ? 
`Make sure to expose the deployed solution on default webserver ports  (80 or 443)`
u mean to create a service to expose that pod on 80 or what?
Also charts in this repo should work behind traefik as a load balancer not node port
check the [ingress part](https://github.com/threefoldtech/vdc-solutions-charts/blob/master/charts/gitea/values.yaml#L30)  and [service part ](https://github.com/threefoldtech/vdc-solutions-charts/blob/master/charts/gitea/values.yaml#L16)
so I recomment that u test it in similar setup like on a vdc or digital ocean machine with k3s installed https://k3s.io/

Comment (tobiaschielens) [reply charts/filebrowser/README.md]: On the jumpscale (with filtst) applications were exposed on an FQDN. The application does a redirect for authorization. When the app is exposed to another port the redirect won't work as the javascript isn't taking into account the port number. If the helm packages will be exposed on an FQDN (port 80/443) there is no problem.

**Comment by tobiaschielens** [comment]
All remarks should be resolved

File: charts/filebrowser/README.md (hunk 3 above)
Comment (ashraffouda) [reply charts/filebrowser/README.md]: can u plz make use of ur ingress.yaml? ingress is responsible for exposing the solution enable it in ur values.yaml 
I need one command to test this chart like the following example
```
helm install monitoring marketplace/monitoring_stack  --set prometheus.ingress.hosts[0]=ashraf-monitoring.webg2dev.grid.tf 
```
after running this command you will get monitoring stack running and exposed on port 443
this section in readme is important to have ssl certificate on ur domain
Screenshot from 2021-01-05 16-57-02
for more details refere to 
https://github.com/threefoldtech/vdc-solutions-charts/blob/master/charts/monitoringstack/values.yaml#L91
https://github.com/threefoldtech/vdc-solutions-charts/blob/master/charts/monitoringstack/values.yaml#L1557
https://github.com/threefoldtech/vdc-solutions-charts/blob/master/charts/monitoringstack/templates/prometheus/ingress.yaml#L14
https://github.com/threefoldtech/vdc-solutions-charts/blob/master/charts/gitea/values.yaml#L16

**Comment by tobiaschielens** [comment]
Added ingress based on https://github.com/threefoldtech/vdc-solutions-charts/blob/master/README.md#note

**Comment by tobiaschielens** [comment]
The following command does work ```helm install monitoring marketplace/monitoring_stack  --set prometheus.ingress.hosts[0]=ashraf-monitoring.webg2dev.grid.tf ```

**Comment by ken-de-moor** [comment]
To be reviewed

**Comment by tobiaschielens** [comment]
Resolved issues, ready for review

**Pull request closed without merging**